skrub/_join_utils.py

```diff
@@ -131,13 +132,7 @@ def check_column_name_duplicates(
         (main_columns, main_table_name),
         (aux_columns, aux_table_name),
     ]:
-        counts = Counter(columns)
-        duplicates = [k for k, v in counts.items() if v > 1]
-        if duplicates:
-            raise ValueError(
-                f"Table '{table_name}' has duplicate column names: {duplicates}."
-                " Please make sure column names are unique."
-            )
+        _utils.check_duplicated_column_names(columns, table_name=table_name)
```

Comment:
I guess the `aux_table.columns` above should be dispatched here

Reply:
as we discussed IRL, the support for polars in joiner will be handled in another pr